src/store/wallet/effects/create/create.ts

```diff
@@ -728,3 +743,94 @@ export const getDecryptPassword =
       );
     });
   };
+
+export const detectAndCreateTokensForEachEvmWallet =
+  ({key, force}: {key: Key; force?: boolean}): Effect<Promise<void>> =>
+  async dispatch => {
+    try {
+      dispatch(
+        LogActions.info('starting [detectAndCreateTokensForEachEvmWallet]'),
+      );
+
+      const evmWalletsToCheck = key.wallets.filter(
+        w =>
+          w.credentials &&
+          IsEVMChain(w.credentials.chain) &&
+          BitpaySupportedEvmCoins[w.credentials.chain] &&
+          w.credentials.coin ===
+            BitpaySupportedEvmCoins[w.credentials.chain].coin &&
+          w.receiveAddress,
+      );
+
+      for (const [index, w] of evmWalletsToCheck.entries()) {
+        if (w.chain && w.receiveAddress) {
+          const erc20WithBalanceData: MoralisErc20TokenBalanceByWalletData[] =
+            await dispatch(
+              getERC20TokenBalanceByWallet({
+                chain: w.chain,
+                address: w.receiveAddress,
+              }),
+            );
+
+          let filteredTokens = erc20WithBalanceData.filter(erc20Token => {
+            // Filter by: token already created in the key (present in w.tokens), possible spam and significant balance
+            return (
+              (!w.tokens ||
+                !cloneDeep(w.tokens).some(token =>
+                  token.includes(erc20Token.token_address),
+                )) &&
+              !erc20Token.possible_spam &&
+              erc20Token.balance &&
+              erc20Token.decimals &&
+              parseFloat(erc20Token.balance) /
+                Math.pow(10, erc20Token.decimals) >=
+                1e-6
+            );
+          });
+
+          let account: number | undefined;
+          let customAccount = false;
+          if (w.credentials.rootPath) {
+            account = getAccount(w.credentials.rootPath);
+            customAccount = true;
+          }
+
+          for (const [index, tokenToAdd] of filteredTokens.entries()) {
+            const newTokenWallet: AddWalletData = {
+              key,
+              associatedWallet: w,
+              currency: {
+                chain: w.chain,
+                currencyAbbreviation: tokenToAdd.symbol.toLowerCase(),
+                isToken: true,
+                tokenAddress: tokenToAdd.token_address,
+              },
+              options: {
+                network: Network.mainnet,
+                ...(account !== undefined && {
+                  account,
+                  customAccount,
+                }),
+              },
+            };
+
+            await dispatch(
```

Comment:
In this case I need the promises to be executed in sequential order and not in parallel, otherwise there is a race in the editing and saving of data within key.